### PR TITLE
#36 Loading adapter scripts from file

### LINKED ISSUE #36: Create adapter script from file
just like for scripts

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #36

### CODE REVIEW

**Review by redcatbear — APPROVED** [review]

**Review by redcatbear — CHANGES_REQUESTED** [review]
Please add an integration test.

File: src/main/java/com/exasol/dbbuilder/dialects/exasol/AdapterScript.java [hunk 1]
```diff
@@ -142,6 +146,30 @@ public Builder content(final String content) {
...
+         * @return {@code this} for fluent programming
+         * @throws IOException in case the file could not be read
+         */
+        public Builder content(final Path path) throws IOException {
+            this.content = Files.readString(path);
+            return this;
+        }
+
+        /**
+         * Set the script content to a jar file from BucketFS.
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/exasol/AdapterScript.java:162]:
```suggestion
         * Set the script content to a JAR file from BucketFS.
```

File: src/test/java/com/exasol/dbbuilder/dialects/exasol/AdapterScriptTest.java [hunk 2]
```diff
@@ -65,6 +70,23 @@ void testGetContent() {
...
+    @Test
+    void testLoadContentFromFile(@TempDir final Path tempDir) throws IOException {
+        final String expected_content = "-- this is a comment";
+        final Path tempFile = tempDir.resolve("script.lua");
+        Files.write(expected_content.getBytes(), tempFile.toFile());
+        assertThat(defaultAdapterScriptBuilder().content(tempFile).build().getContent(), equalTo(expected_content));
+    }
+
+    @Test
+    void testBucketFsContent() {
```
Comment (redcatbear) [inline src/test/java/com/exasol/dbbuilder/dialects/exasol/AdapterScriptTest.java:82]: :+1:

File: src/test/java/com/exasol/dbbuilder/dialects/exasol/ExasolSchemaTest.java [hunk 3]
```diff
@@ -36,6 +36,14 @@ void testCreateTableBuilder() {
         assertThat(table.getName(), equalTo("TABLE_D"));
     }
 
+    @Test
+    void testCreateAdapterScriptBuilder() {
```
Comment (redcatbear) [inline src/test/java/com/exasol/dbbuilder/dialects/exasol/ExasolSchemaTest.java:40]: The name of this method does not fit the test inside.

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/dbbuilder/dialects/exasol/ExasolSchemaTest.java (hunk 3 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/dbbuilder/dialects/exasol/ExasolSchemaTest.java:40]: why not? It tests the method `exasolSchema.createAdapterScriptBuilder()`

File: src/main/java/com/exasol/dbbuilder/dialects/exasol/ExasolImmediateDatabaseObjectWriter.java [hunk 4]
```diff
@@ -42,8 +42,7 @@ public void write(final AdapterScript adapterScript) {
                         + "You can either disable debugging or provide a debug connection with the AdatperScript creation.");
             }
             final String debuggerConnection = adapterScript.getDebuggerConnection();
-            sqlBuilder.append("-agentlib:jdwp=transport=dt_socket,server=n,address=").append(debuggerConnection)
-                    .append("\n");
+            sqlBuilder.append("%jvmoption  -agentlib:jdwp=transport=dt_socket,server=n,address=" + debuggerConnection+";\n");
```
Comment (redcatbear) [inline src/main/java/com/exasol/dbbuilder/dialects/exasol/ExasolImmediateDatabaseObjectWriter.java:45]:
```suggestion
            sqlBuilder.append("%jvmoption -agentlib:jdwp=transport=dt_socket,server=n,address=" + debuggerConnection+";\n");
```

**Review by redcatbear — DISMISSED** [review]
Approved with one editorial change

**Review by redcatbear — APPROVED** [review]

**Pull request merged**